lib/util/jsdoc.js

```diff
@@ -33,10 +33,15 @@ class MethodDef {
   constructor() {
     this.params = []
     this.tags = {}
+    this.argsData = {}
   }
 
-  addParam(name, type, optional, description) {
-    this.params.push(new ParamDef(name, type, optional, description))
+  addParam(name, type, optional, description, dictionary, dependsOn) {
+    this.params.push(new ParamDef(name, type, optional, description, dictionary, dependsOn))
+
+    if (dictionary || dependsOn) {
+      this.argsData[name] = { dictionary, dependsOn }
+    }
```

Comment:
I'm not sure, but looks like

`argsData` is not the best decision. We could do something like:

```
constructor() {
    this.params = []
    this.tags = {}
    this.methodData = {
       args: {}
    }
  }

  addParam(name, type, optional, description, dictionary, dependsOn) {
    this.params.push(new ParamDef(name, type, optional, description, dictionary, dependsOn))

    if (dictionary || dependsOn) {
      this.methodData.args[name] = { dictionary, dependsOn }
    }
  }
```

And inside `methodData` set anything what we want in the future, not only arguments data


It's in case if I get your idea correctly :sweat_smile:

Reply:
agree, only I would avoid using the word `method` since it's already in a method object context, `methodTags` `methodParams` sound redundant 

what about calling it `meta`, `metaData`, `metaInfo` ? btw, I assume we already have `metaInfo` for some entity, perhaps for the service object the server returns the property